Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -415,10 +415,10 @@ class SourceGenerator {
         let createIntermediateGroups = targetSource.createIntermediateGroups ?? project.options.createIntermediateGroups
 
         let directories = children
-            .filter { $0.isDirectory && $0.extension == nil && $0.extension != "lproj" }
+            .filter { $0.isDirectory && $0.extension != "lproj" && $0.extension != "xcassets" && $0.extension != "bundle" && $0.extension != "xcdatamodeld" }
```

Comment:
The extension == nil, would have always prevented the next && from being evaluated.

So I have removed that and explicitly whitelisted the required folders. Most of these extensions are something I have discovered after trying to fix broken unit tests